### PR TITLE
feat(/xref): add UI for xref search

### PR DESCRIPTION
**Pull request by sidvishnoi**
Depends upon https://github.com/marcoscaceres/respec.org/pull/23

### CODE REVIEW

File: static/xref/index.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
```
Comment (marcoscaceres) [inline static/xref/index.html]: We don’t want to encourage them ;) 
```suggestion
```

File: static/xref/index.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    <title>Xref UI</title>
```
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
    <title>Search cross references</title>
```

File: static/xref/index.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    <link rel="stylesheet" href="style.css">
+  </head>
+  <body>
+    <main>
```
Comment (marcoscaceres) [inline static/xref/index.html]: Not needed, as we only have main content and no auxiliary stuff. 
```suggestion
```

File: static/xref/index.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      <form action="/xref" id="xref-search" name="xref-search" method="post">
+        <div class="field">
+          <label for="term">Term</label> <input name="term" placeholder=
+          "target">
+        </div>
+        <div class="field">
+          <label for="cite">Shortnames</label>
```
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
          <label for="cite">Specification(s) short name</label>
```

File: static/xref/index.html [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Xref UI</title>
+    <link rel="stylesheet" href="style.css">
+  </head>
+  <body>
+    <main>
+      <form action="/xref" id="xref-search" name="xref-search" method="post">
+        <div class="field">
+          <label for="term">Term</label> <input name="term" placeholder=
```
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
          <label for="term">Terms, concept, or WebIDL</label> <input name="term" placeholder=
```

File: static/xref/index.html [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+          "target">
```
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
          "EventTarget, event target">
```

File: static/xref/index.html [hunk 7]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
```
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
    <meta charset="UTF-8">
```

File: static/xref/index.html [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
```

File: static/xref/index.html (hunk 1 above)
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
```

File: static/xref/index.html [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+          "ResizeObservation">
+      </form>
+      <section id="output"></section>
+      <template id="table">
+        <table>
+          <thead>
+            <tr>
+              <th>Spec</th>
+              <th>Type</th>
+              <th>URI</th>
```
Comment (marcoscaceres) [inline static/xref/index.html]: Definitely don’t want to show this, as we want to discourage editors from using these. 
```suggestion
```

We should link to the spec instead, allowing editors to go read the definition.

File: static/xref/index.html [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+          placeholder="dfn,interface,_IDL_">
+        </div>
+        <div class="field">
+        </div>
+        <button>Search</button>
+          <label for="for">For</label> <input name="for" placeholder=
+          "ResizeObservation">
+      </form>
+      <section id="output"></section>
+      <template id="table">
```
Comment (marcoscaceres) [inline static/xref/index.html]: Template only the table body, the rest can stay. We should have an initial placeholder with some instructions. Also, if no results are found, the table body should say “no results found”.

File: static/xref/index.html [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+              <th>Spec</th>
+              <th>Type</th>
+              <th>URI</th>
+              <th>for</th>
+            </tr>
+          </thead>
+          <tbody></tbody>
+        </table>
+      </template>
+    </main>
```
Comment (marcoscaceres) [inline static/xref/index.html]:
```suggestion
```

File: static/xref/index.html [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    <script src="script.js"></script>
```
Comment (marcoscaceres) [inline static/xref/index.html]: Move to head and do: 
```suggestion
    <script src="script.js" defer></script>
```

File: static/xref/index.html (hunk 9 above)
Comment (sidvishnoi) [reply static/xref/index.html]: We'll need access to specref here, right? To convert shortname to full URL. OR should we handle it in backend?

File: static/xref/index.html (hunk 10 above)
Comment (sidvishnoi) [reply static/xref/index.html]: I think we should use a `<ul><li>` instead of table.

File: static/xref/index.html [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+      <button>Search</button>
+    </form>
+    <section id="output"></section>
+    <template id="table">
+      <table>
+        <thead>
+          <tr>
+            <th>Spec</th>
+            <th>Type</th>
+            <th>for</th>
```
Comment (marcoscaceres) [inline static/xref/index.html]: I’m thinking we need a last row that shows how to cite. Like, if it’s IDL, {{WhatEver}}, if it’s a “concept for”, [= list/for each =], and so on. Then Editors just copy paste the right thing.

File: static/xref/index.html [hunk 14]
```diff
@@ -16,7 +16,7 @@
       <div class="field">
         <label for="cite">Specification(s) short name</label>
         <div class="selections" data-for="cite"></div>
-        <input name="cite" placeholder="html,webidl,payment-request" />
+        <input name="cite" placeholder="html,webidl,infra..." />
```
Comment (sidvishnoi) [inline static/xref/index.html]: these placeholders get overridden as soon as `/xref/meta` resolves and we replace this `input` with `input[is]` though. but good to have it here.

File: static/xref/script.js [hunk 15]
```diff
@@ -91,6 +91,9 @@ function getFormData() {
 async function onSubmit(event) {
   event.preventDefault();
   const data = getFormData();
+  if (data.term === "") {
```
Comment (sidvishnoi) [inline static/xref/script.js]: does it really crash? `""` is a valid enum value, right? IIRC, it was defined at 3 places

Comment (marcoscaceres) [reply static/xref/script.js]: Sorry, this was before I realized that I was missing PR 34 from the other repo.

**Pull request merged**